### ISSUE TITLE
1- Allow to transfer an individual Sample to another institution

### ISSUE THREAD

**Issue by diegoliberman** [instedd/cdx#1385]
PERSONA: I'm Diego, and I work at a health/research organization.  Currently I'm in charge of a program that involves distributing hundreds of our samples, to a group of laboratories that will be running some tests.

As a program coordinator, I want to transfer the Samples to the recipient institution, so that they can access the information they need, and there's some tracking data kept for future reference.

**Constraint**: this work item doesn't consider transferring samples that belong to a batch

### Acceptance criteria

- [x] 0. You shouldn't start working on this effort, unless you acknowledge what is mentioned in 3.2, 3.4 and 3.5 in this issue.
- [x] 1. In the Samples view, I can see an action icon next to the print icon.

https://projects.invisionapp.com/d/main#/console/4103589/453801736/preview
![image](https://user-images.githubusercontent.com/25181052/146451524-b85a8887-602d-4a72-9c32-ee40d93d060f.png)

- [x] 2. The transfer icon is disabled, unless there is at least 1 sample selected.  Just like the print action.

- [x] 3. After pressing the Transfer icon, a modal gets opened.
https://projects.invisionapp.com/d/main#/console/4103589/461729946/preview

#### In the modal
- [x] 3.1. I see a list of samples that were selected for transferring.
- [x] 3.2. The feature for removing samples is out of scope.  We'll do that some date later in #1386.  Please mark this check box to acknowledge that you won't implement this now.
- [x] 3.3. I see a dropdown with an autocomplete for picking 1 institution.  Leverage the learnings of the spike #1401.  
- [x] 3.3.1- The list of institutions doesn't contain the institution I'm currently logged in with.
- [x] 3.4. Files.  This will be done as part of other ticket (#1387).  Please mark this check box to acknowledge that you won't implement this now.
- [x] 3.5. Notes. This will be done as part of other ticket (#1387).  Please mark this check box to acknowledge that you won't implement this now.
- [x] 3.6. I see a `Include QC data` checkbox.   This will have no effect until we implement a future story for transferring samples that belong to a batch. 
- [x] 3.7. I see a cancel action.  When it gets pressed the modal is closed.
- [x] 3.8. I see a Send action

- [x] 4.1 The send action changes the access of the Sample, to make it belong to the recipient institution.
- [x] 4.2 If the sample being processed belongs to a batch, do nothing for that Sample.  We'll take care of this use case with another issue.

5. [x] I see a nice success message
https://projects.invisionapp.com/d/main#/console/4103589/461729949/preview
![image](https://user-images.githubusercontent.com/25181052/146453135-a30a1da0-be44-4ce7-9633-4a8fc6ac7954.png)

6. If I go back to the samples list, the ones that I have transferred don't appear anymore.
7. If I login with a user of the recipient institution, I see the transferred samples in the list of samples.

### Test cases
- [x] After transferring Samples, make sure that it appears in the list of samples of the recipients, and that it disappear from the sender's list.
- [x] Just in case, test transferring just 1 sample, and then test transferring more than 1.
- [x] Try to transfer a Sample more than once.  That means that a recipient, transfer the sample again to another institution. 
- [x] Test to transfer a sample back to the sender Institution.
- [x] If the Sample used to belong to a Site within the sender Institution, the sample gets sent to the recipient institution, without within a particular site.

**Comment by diegoliberman** [comment]
@gbatiston transfer will be done to Institutions.  Not to sites within the institution, which would be out of scope right now.

**Comment by diegoliberman** [comment]
Verified in `Version: dev-525a860 (build 1058)`, but the success message toast didn't appear and was reported in #1528.
